COMPILING.md

```diff
@@ -282,6 +283,92 @@ To build full curses version with localizations and lua enabled:
+  
+Some dependencies need their entire source code copied under the Android project folder so they can be built for Android - see below.
+
+### Setup
+
+Install Linux dependencies. For a desktop Ubuntu installation:
+
+    sudo apt-get install lua5.2 openjdk-8-jdk-headless
+
+Install Android SDK and NDK:
+
+    wget https://dl.google.com/android/repository/sdk-tools-linux-4333796.zip
+    unzip sdk-tools-linux-4333796.zip -d ~/android-sdk
+    rm sdk-tools-linux-4333796.zip
+    ~/android-sdk/tools/bin/sdkmanager --update
+    ~/android-sdk/tools/bin/sdkmanager "tools" "platform-tools" "ndk-bundle"
+    ~/android-sdk/tools/bin/sdkmanager --licenses
+
+Export Android environment variables (you can add these to the end of `~/.bashrc`):
+
+    export ANDROID_SDK_ROOT=~/android-sdk
+    export ANDROID_HOME=~/android-sdk
+    export ANDROID_NDK_ROOT=~/android-sdk/ndk-bundle
+    export PATH=$PATH:$ANDROID_SDK_ROOT/platform-tools
+    export PATH=$PATH:$ANDROID_SDK_ROOT/tools
+    export PATH=$PATH:$ANDROID_NDK_ROOT
+
+Install custom Android dependencies. Change directory to `Cataclysm-DDA` (the root folder of this repository) and run:
+
+    git clone https://github.com/a1studmuffin/SDL2 ./android/app/jni/SDL2
+    git clone https://github.com/a1studmuffin/lua ./android/app/jni/lua
+    git clone https://github.com/a1studmuffin/libintl-lite ./android/app/jni/libintl-lite
+
+Install other Android dependencies SDL2_ttf, SDL2_mixer and SDL2_image. You may download the release ZIPs from from https://hg.libsdl.org/ and install them manually under `./android/app/jni/SDL2_xxx`, or clone latest dev with hg:
+
+    hg clone http://hg.libsdl.org/SDL_ttf ./android/app/jni/SDL2_ttf
+    hg clone http://hg.libsdl.org/SDL_mixer ./android/app/jni/SDL2_mixer
+    hg clone http://hg.libsdl.org/SDL_image ./android/app/jni/SDL2_image
+
+Manually generate version and lua bindings:
+
+    make version
+    cd src/lua && lua generate_bindings.lua
+
+### Android device setup
+
+Enable [Developer options on your Android device](https://developer.android.com/studio/debug/dev-options). Connect your device to your PC via USB cable and run:
+
+    adb devices
+    adb connect <devicename>
+
+### Building
+
+To build an APK, use the Gradle wrapper command line tool (gradlew). The Android Studio documentation provides a good summary of how to [build your app from the command line](https://developer.android.com/studio/build/building-cmdline).
+
+To build a debug APK, from the `android/` subfolder of the repository run:
+
+    ./gradlew assembleDebug
+
+This creates a debug APK in `./android/app/build/outputs/apk/` ready to be installed on your device.
+
+To build a debug APK and immediately deploy to your connected device over adb run:
+
+    ./gradlew installDebug
+
+To build a signed release APK (ie. one that can be installed on a device), [build an unsigned release APK and sign it manually](https://developer.android.com/studio/publish/app-signing#signing-manually).
```

Comment:
Might be useful to replicate commands from instruction from the link to create key, align apk and sign it.

Also had to install additional utilities to align apk zip and sign it:

```
sudo apt-get install zipalign
sudo apt-get install apksigner
```